src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentBindLoweringPass.cs

```diff
@@ -78,18 +131,42 @@ private void ProcessDuplicates(IntermediateNode node)
             {
                 // For each usage of the general 'fallback' bind tag helper, it could duplicate
                 // the usage of a more specific one. Look for duplicates and remove the fallback.
-                var attribute = node.Children[i] as TagHelperPropertyIntermediateNode;
+                TagHelperDescriptor tagHelper = null;
+                string attributeName = null;
+                var attribute = node.Children[i];
+                if (attribute is TagHelperPropertyIntermediateNode propertyAttribute)
+                {
+                    attributeName = propertyAttribute.AttributeName;
+                    tagHelper = propertyAttribute.TagHelper;
+                }
+                else if (attribute is TagHelperAttributeParameterIntermediateNode parameterAttribute)
+                {
+                    attributeName = parameterAttribute.AttributeName;
+                    tagHelper = parameterAttribute.TagHelper;
+                }
                 if (attribute != null &&
-                    attribute.TagHelper != null &&
-                    attribute.TagHelper.IsFallbackBindTagHelper())
+                    tagHelper != null &&
+                    tagHelper.IsFallbackBindTagHelper())
```

Comment:
You'll probably want to generalize this de-duplification so it can be used for more than just `bind`. If you want to do that later to get this rolling that's fine too.

Reply:
I'm not sure how. Could you elaborate? This is what I was talking about when I said we still needed to special case some stuff. This deduplication logic is very specific to `bind` and how its descriptors are configured.

Reply:
Maybe I don't have enough context but I thought it'd also be a common pattern for parameters to overlap with non-parameter bound attributes (`bind-*` overlapping with `bind-*:*`; in which case de-duping the IR bits would be needed for all similar attributes? If i'm wrong here feel free to ignore 😄

Reply:
That is not what is happening here. It is deduping cases like, the catch-all(fallback) `bind-*` bound attribute descriptor can match along with a more specific `bind-value` bound attribute descriptor for the same attribute like `bind-value="..."` in which case we want to remove the former. I just updated it to do the same for parametered attributes. (remove `bind-*:blah` and keep `bind-value:blah`).